content/post/lessons-learned-building-an-accessible-web-application-framework/index.md

```diff
@@ -0,0 +1,69 @@
+
+#### Impaired cognition
+
+Focus on applying layout simplicity to our workflows and components as well as improving keyword comprehension by emphasizing common design patterns and expected behaviors. For example, buttons should not navigate a user to a new page, that is the role of links.
+
+#### All impairments
+
+Provide a consistent semantic layout and access to information regardless of form factor (desktop, mobile). Tooltips are a good example of this. On desktop tooltips are accessed via hover, where as on mobile, the hover event doesn't exist. The information contained in the tooltip has to be offered in a way that can be accessed on both devices.
+
+### Design for accessibility
+
+Accessibility cannot simply be layered on any component as an afterthought. This was a mistake we made early on. We’d work through designing and building a complex component and only test screen reader compatibility at the end. Unsurprisingly, the component would fail, and we’d have to repeat the process again to correct the issues, sometimes with a complete redesign. Just as engineering constraints inform a design, so to do accessibility constraints. Designing for them up front will reduce expensive re-work late in a project.
+
+As you can see, there is far more about accessibility to consider than ARIA labels. We understand that implementing accessibility on the web is a complex and difficult task and we've built the Terra application framework to help.
+
+## What does Terra offer for accessibility
+
+Terra provides a stable base on which to build accessible applications. The Terra application framework and each Terra component are designed and built to meet our [accessibility standards](https://engineering.cerner.com/terra-ui/about/terra-ui/component-standards#accessibility-a11y). To help ensure that our components meet these standards, we perform a mixture manual testing and automated testing using axe-core.
+
+While using Terra gives you a good head start at building an accessible application, simply by using Terra you cannot assume your application is fully accessible. How Terra components are assembled and labeled in an application have a large impact on its accessibility. Any custom components need to be vetted with the same care as Terra components to ensure accessibility. Application owners should make sure to test their apps using a screen reader.
+
+Developing applications for the web is inherently complex. Adding accessibility does not make development simpler. However, we've found that by ensuring accessibility for the Terra application framework and components, not only are we ensuring our apps are available to all users, but we also build better architected, more predictable products. It has been our experience that implementing a consistent semantic layout and a sensible keyboard navigation order lead directly to a logically organized information architecture.
```

Comment:
> However, we've found that by ensuring accessibility for the Terra application framework and components, not only are we ensuring our apps are available to all users, but we also build better architected, more predictable products.

This is pretty lengthy sentence. Perhaps:

> By ensuring accessibility is applied to the Terra application framework and components, we are making the right thing, the easy thing to do for applications utilizing Terra. Furthermore, this has made our products more predictable in how users access their functionality since it is being consistently applied.